### PR TITLE
Task-35993: Send onboarding email

### PR DESCRIPTION
**Pull request by azayati**

### CODE REVIEW

File: component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryService.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
     Credentials verifyToken(String tokenId);
     boolean changePass(final String tokenId, final String username, final String password);
     public boolean sendRecoverPasswordEmail(User user, Locale defaultLocale, HttpServletRequest req);
+    public boolean sendOnbordingEmail(User user, Locale defaultLocale, StringBuilder url);
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryService.java]:
```suggestion
    public boolean sendOnboardingEmail(User user, Locale defaultLocale, StringBuilder url);
```

File: component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java [hunk 2]
```diff
@@ -94,6 +97,64 @@ public boolean changePass(final String tokenId, final String username, final Str
             return false;
         }
     }
+    
+    @Override
+    public boolean sendOnbordingEmail(User user, Locale defaultLocale, StringBuilder url) {
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java]:
```suggestion
    public boolean sendOnboardingEmail(User user, Locale defaultLocale, StringBuilder url) {
```

File: component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java [hunk 3]
```diff
@@ -94,6 +97,64 @@ public boolean changePass(final String tokenId, final String username, final Str
...
+        ResourceBundle bundle = bundleService.getResourceBundle(bundleService.getSharedResourceBundleNames(), locale);
+
+        Credentials credentials = new Credentials(user.getUserName(), "");
+        String tokenId = remindPasswordTokenService.createToken(credentials);
+
+        url.append("/on-bording");
+        url.append("?lang=" + I18N.toTagIdentifier(locale));
+        url.append("?token=" + tokenId);
+        String emailBody = buildOnbordingEmailBody(user, bundle, url.toString());
+        String emailSubject = bundle.getString("onbording.email.header") + " " + brandingService.getCompanyName();
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java]:
```suggestion
        String emailSubject = bundle.getString("onboarding.email.header") + " " + brandingService.getCompanyName();
```

File: component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java [hunk 4]
```diff
@@ -94,6 +97,64 @@ public boolean changePass(final String tokenId, final String username, final Str
...
+            mailService.sendMessage(message);
+        } catch (Exception ex) {
+            log.error("Failure to send onbording email", ex);
+            return false;
+        }
+
+        return true;
+    }
+    
+    private String buildOnbordingEmailBody(User user, ResourceBundle bundle, String link) {
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java]:
```suggestion
    private String buildOnboardingEmailBody(User user, ResourceBundle bundle, String link) {
```

File: component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java [hunk 5]
```diff
@@ -94,6 +97,64 @@ public boolean changePass(final String tokenId, final String username, final Str
...
+      String content;
+      InputStream input = this.getClass().getClassLoader().getResourceAsStream("conf/onBording_email_template.html");
```
Comment (plamarque) [inline component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryServiceImpl.java]:
```suggestion
      InputStream input = this.getClass().getClassLoader().getResourceAsStream("conf/onBoarding_email_template.html");
```

File: component/web/security/src/main/resources/conf/onBording_email_template.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                    <td align="center" style="padding: 10px 0 10px">
+                        <table width="90%" cellpadding="0" cellspacing="0" >
+                            <tr>
+                                <td align="left" style="line-height: 18px;font-size: 13px;color:#333333;font-family:HelveticaNeue,helvetical,arial;">
+                                    <p style="margin: 0 0 10px 0;line-height: 22px">&{onbording.email.hello} ${USER_DISPLAY_NAME},</p>
+                                    <p style="margin: 0 0 10px 0;line-height:22px">&{onbording.email.welcome} ${COMPANY_NAME}</p>
+                                    <table width="100%" cellpadding="0" cellspacing="0">
+                                        <tr>
+                                            <td bgcolor="#f9f9f9" style="padding: 10px">
+                                                <p style="margin: 0 0 10px 0;line-height:22px">&{onbording.email.accessLink}: <a href="${RESET_PASSWORD_LINK}" style="text-decoration: none;color:#2e5d97">&{onbording.email.join}</a></p>
```
Comment (plamarque) [inline component/web/security/src/main/resources/conf/onBording_email_template.html]:
```suggestion
                                                <p style="margin: 0 0 10px 0;line-height:22px">&{onboarding.email.accessLink}: <a href="${RESET_PASSWORD_LINK}" style="text-decoration: none;color:#2e5d97">&{onboarding.email.join}</a></p>
```

File: component/web/security/src/main/resources/conf/onBording_email_template.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                                            </td>
+                                        </tr>
+                                    </table>
+                                    <p style="margin: 10px 0 10px 0;line-height:22px">&{onbording.email.bye}.</p>
```
Comment (plamarque) [inline component/web/security/src/main/resources/conf/onBording_email_template.html]:
```suggestion
                                    <p style="margin: 10px 0 10px 0;line-height:22px">&{onboarding.email.bye}.</p>
```

File: component/web/security/src/main/resources/conf/onBording_email_template.html [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                            </tr>
+                        </table>
+                    </td>
+                </tr>
+                <!--end body-->
+
+                <!-- begin footer-->
+                <tr>
+                    <td height="38" valign="middle" align="center" bgcolor="#486694" style="color:#fff;font-family:HelveticaNeue,helvetical,arial;">
+                        <h2 style="color: #fff;margin: 0;padding: 0; font-size: 13px;">&{onbording.email.email.footer}</h2>
```
Comment (plamarque) [inline component/web/security/src/main/resources/conf/onBording_email_template.html]:
```suggestion
                        <h2 style="color: #fff;margin: 0;padding: 0; font-size: 13px;">&{onboarding.email.email.footer}</h2>
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 9]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
 gatein.forgotPassword.email.linkDescription=To reset your password, simply click the link below.
 gatein.forgotPassword.email.helpMessage=If you do not want to change your password, please ignore this email.
 gatein.forgotPassword.email.footer=eXo Platform
+###### Onbording feature
+onbording.email.subject=Invitation to join the platform
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.subject=Invitation to join the platform
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 10]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.header=Invitation to join the platform
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.header=Invitation to join the platform
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 11]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.hello=Hello
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.hello=Hello
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 12]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.welcome=You are invited to join the platform
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.welcome=You are invited to join the platform
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 13]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.accessLink=To access it click on the following link
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.accessLink=To access it click on the following link
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 14]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.join=Join
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.join=Join
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 15]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.bye=Thank you and very nice collaboration within your community!
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.bye=Thank you and very nice collaboration within your community!
```

File: web/portal/src/main/resources/locale/portal/webui_en.properties [hunk 16]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Your username
...
+onbording.email.email.footer=eXo Platform
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_en.properties]:
```suggestion
onboarding.email.email.footer=eXo Platform
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 17]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
 gatein.forgotPassword.email.linkDescription=Pour r\u00E9initialiser votre mot de passe, cliquez simplement sur le lien ci-dessous.
 gatein.forgotPassword.email.helpMessage=Si vous ne voulez pas changer votre mot de passe, veuillez ignorer ce message.
 gatein.forgotPassword.email.footer=eXo Platform
+###### Onbording feature
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
###### Onboarding feature
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 18]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.subject=Invitation à rejoindre la platforme
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.subject=Invitation à rejoindre la plateforme
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 19]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.header=Invitation à rejoindre la platforme
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.header=Invitation à rejoindre la plateforme
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 20]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.hello=Bonjour
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.hello=Bonjour
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 21]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.welcome=Vous êtes invité à rejoindre la plateforme
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.welcome=Vous êtes invité(e) à rejoindre la plateforme
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 22]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.accessLink=Pour y accéder, veillez cliquer sur le lien suivant
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.accessLink=Pour y accéder, veuillez cliquer sur le lien suivant
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 23]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.join=Rejoindre
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.join=Rejoindre
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 24]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.bye=Merci et très belle collaboration au sein de votre communauté!
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.bye=Merci et très belle collaboration au sein de votre communauté!
```

File: web/portal/src/main/resources/locale/portal/webui_fr.properties [hunk 25]
```diff
@@ -1333,3 +1333,12 @@ gatein.forgotPassword.email.username=Votre nom d'utilisateur
...
+onbording.email.email.footer=eXo Platform
```
Comment (plamarque) [inline web/portal/src/main/resources/locale/portal/webui_fr.properties]:
```suggestion
onboarding.email.email.footer=eXo Platform
```

**Review by plamarque — DISMISSED** [review]
typos.
Also don't forget to rename the email template file

**Comment by azayati** [comment]
@plamarque All done !

**Review by rdenarie — DISMISSED** [review]
LGTM

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**